src/Controller/FeedController.php

```diff
@@ -38,18 +39,19 @@ public function index(DogCRUDService $dogService, PostRepository $postRepository
         );
     }
 
-    public function createComment(Request $request, PostCRUDService $postService) : Response
+    public function createComment(Request $request, EntityManagerInterface $em , Post $post) : Response
     {
         $comment = new Comment();
-
+        $comment->setPost($post);
         $form = $this->createForm(CommentFormType::class, $comment);
 
         $dogUser = $this->getUser();
+        $comment->setDog($dogUser);
         $form->handleRequest($request);
         
         if($form->isSubmitted() && $form->isValid()) {
-            dd("needs more work :)");
-            // $postService->saveCommentForPost($dogUser, $comment, $post);
+            $em->persist($comment);
+            $em->flush();
             return $this->redirectToRoute('feed');
```

Comment:
I think this should be in a service. Don't forget: controllers should be lean and not contain details of what is happening. Also, we should be using a repository rather than using the entity manager directly.

Reply:
I've made this change to the postCRUDService, make the Post handle all the comment operations?